lib/index.d.ts

```diff
@@ -0,0 +1,648 @@
+   */
+  authorizationUrl (parameters?: IAuthorizationUrlParams): string
+
+  /**
+   * Returns the target logout redirect URI to redirect End-Users to using the provided parameters.
+   * @param parameters
+   */
+  endSessionUrl (parameters?: IEndSessionUrlParams): string
+
+  /**
+   * Returns recognized callback parameters from a provided input.
+   *
+   * - When input is of type string it will be parsed using url.parse and its query component will be returned
+   * - When input is a GET http/http2 request object its url property will be parsed using url.parse and its
+   * query component will be returned
+   * - When input is a POST http/http2 request object its body property will be parsed or returned if it is already
+   * an object. Note: the request read stream will not be parsed, it is expected that you will have a body parser
+   * prior to calling this method. This parser would set the req.body property
+   */
+  callbackParams (input: string | IncomingMessage | Http2ServerRequest): CallbackParamsType
+
+  /**
+   * Performs the callback for Authorization Server's authorization response.
+   * @param redirectUri redirect_uri used for the authorization request
+   * @param parameters returned authorization response, see client.callbackParams if you need help getting them.
+   * @param checks
+   * @param extras
+   */
+  callback(redirectUri: string | undefined, parameters: CallbackParamsType, checks?: IExtendedCallbackChecks, extras?: ICallbackExtras): Promise<TokenSet>
+
+  /**
+   * Pure OAuth 2.0 version of callback().
+   * @param redirectUri
+   * @param parameters
+   * @param checks
+   * @param extras
+   */
+  oauthCallback(redirectUri: string | undefined, parameters: CallbackParamsType, checks?: ICallbackChecks, extras?: ICallbackExtras): Promise<TokenSet>
+
+  /**
+   * Performs refresh_token grant type exchange.
+   * @param refreshToken Refresh Token value. When TokenSet instance is provided its refresh_token property
+   * will be used automatically.
+   * @param extras
+   */
+  refresh(refreshToken: TokenType, extras?: IRefreshExtras): Promise<TokenSet>
+
+  /**
+   * Fetches the OIDC userinfo response with the provided Access Token. Also handles signed and/or
+   * encrypted userinfo responses. When TokenSet is provided as an argument the userinfo sub property
+   * will also be checked to match the on in the TokenSet's ID Token.
+   *
+   * @param accessToken Access Token value. When TokenSet instance is provided its access_token property
+   * will be used automatically.
+   */
+  userinfo(accessToken: TokenType): Promise<object>
+
+  /**
+   * Performs an arbitrary grant_type exchange at the token_endpoint.
+   */
+  grant(body: IGrantBody, extras?: IGrantExtras): Promise<TokenSet>
+
+  /**
+   * Introspects a token at the Authorization Server's introspection_endpoint.
+   */
+  introspect(token: string, tokenTypeHint?: string, extras?: IIntrospectExtras): Promise<object>
```

Comment:
Here a more detailed response type would bee cool.

For example `active: boolean` is a required response property. See  https://tools.ietf.org/html/rfc7662#section-2.2.

Maybe just copy [`IntrospectionResponse`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/d8907f303d75056e0710a72bad713179fbc78240/types/openid-client/index.d.ts#L93)?

Reply:
better now?